### PR TITLE
Datatypes: charging points

### PR DESCRIPTION
**Pull request by mdavezac**
Creates charging point data structure.

At this point, the datastructure is just an pandas.DataFrame with longitude, latitude, socket type and charger type. Optionally, it can be a dask dataframe. I'm not completely sure whether that last point is good idea or not. We may want to remove it once we have a better idea of what algorithms this goes into.

The PR also sets up some of the infrastructure, especially with regards to checking doctests. See "docs/source/installation.md:Testing."

Part of #8 and #4.

### CODE REVIEW

**Comment by mdavezac** [comment]
@cc-a, this is ready for review.

File: src/evosim/supply.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    
+    import pandas as pd
+    import numpy as np
+    import dask.dataframe as dd
+    import evosim
+
+    pd.set_option('precision', 2)
+
+Charging points are a list of locations (latitude and longitude) with a socket and
+charger type. The simplest way to generate one is to use
```
Comment (cc-a) [inline src/evosim/supply.rst]: Saying list here makes me look for a Python list.
```suggestion
Charging points are a pair of location values (latitude and longitude) with a
socket and charger type. The simplest way to generate one is to use
```

File: docs/source/api.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Developer API catalogue
+=======================
+
+Supply
+======
+
+Enumerations
+------------
+
+.. autoclass:: evosim.supply.ChargerTypes
```
Comment (cc-a) [inline docs/source/api.rst]: There is a warning for this in the output of `sphinx-build`

```suggestion
.. autoclass:: evosim.supply.Chargers
```

File: docs/source/api.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    :members:
+    :undoc-members:
+
+.. autoclass:: evosim.supply.SocketTypes
```
Comment (cc-a) [inline docs/source/api.rst]: As above
```suggestion
.. autoclass:: evosim.supply.Sockets
```

File: src/evosim/supply.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        charger_types: A list of :py:class:`~evosim.supply.Chargers` from which to
+            choose randomly. Defaults to all available charger types.
+        seed: seed for the random number generators. Defaults to ``None``. See
+            :py:func:`numpy.random.default_rng`.
+        **kwargs: If keywords are given, then they should be those of
+            :py:func:`dask.dataframe.from_pandas`
+
+    Returns:
+        Union[dask.dataframe.DataFrame, pandas.DataFrame]: If no keyword arguments are
+        givent, then the funtion returns a :py:class:`pandas.DataFrame`. Otherwise, it
```
Comment (cc-a) [inline src/evosim/supply.py]:
```suggestion
        given, then the funtion returns a :py:class:`pandas.DataFrame`. Otherwise, it
```

**Review by cc-a — APPROVED** [review]
Looks good. A few minor tweaks. You might may also want to add the `-W` flag to sphinx-build to catch the documentation warnings during CI.

**Pull request merged**